### PR TITLE
52 refactor dao for bounded vec

### PR DESCRIPTION
**Pull request by f-gate**
Changed the use of Vec<T> to BoundedVec<T> throughout the dao pallet.

### CODE REVIEW

**Comment by f-gate** [comment]
also pulled in changes from 69-fix release bugs

File: pallets/dao/src/mock.rs [hunk 1]
```diff
@@ -98,13 +98,26 @@ parameter_types! {
 	pub const MaxDaoNameLen: u32 = 64;
 	#[derive(TypeInfo, MaxEncodedLen, Encode)]
 	pub const MaxVisionLen: u32 = 64;
+	#[derive(TypeInfo, MaxEncodedLen, Encode)]
+	pub const MaxMemberOfLen: u32 = 64;
+	#[derive(TypeInfo, MaxEncodedLen, Encode)]
+	pub const MaxMembersPerOrganisation: u32 = 10000;
```
Comment (stojanov-igor) [inline pallets/dao/src/mock.rs]: `MaxMembersPerOrganisation` = Dunbar number.
`MaxOrganisationsPerMember` = Dunbar number
`MaxOrganisationsPerMember` = Dunbar number

File: pallets/dao/src/lib.rs [hunk 2]
```diff
@@ -287,8 +304,14 @@ pub mod pallet {
 		OrganizationAlreadyExists,
 		/// The user is not a member of this organization.
 		NotMember,
+		/// The user if over the maximum amount of organizations allowed to be affiliated with.
```
Comment (stojanov-igor) [inline pallets/dao/src/lib.rs]: We use the comments above the error message to display to the user. They should be descriptive of what the end user sees in the UI. 

For example:
/// You are not allowed to join more organizations at this time.

File: pallets/dao/src/lib.rs [hunk 3]
```diff
@@ -219,22 +236,22 @@ pub mod pallet {
...
-	/// Storage Map to indicate which user agree with a proposed Vision of an Organisation [DaoId, Vec[Account]]
-	pub(super) type ApplicantsToOrganization<T: Config> = StorageMap<_, Twox64Concat, OrganizationIdOf<T>, Vec<T::AccountId>, ValueQuery>;
+	/// Storage Map to indicate which user agree with a proposed Vision of an Organisation [OrganizationId, BoundedVec[Account]]
+	pub(super) type ApplicantsToOrganization<T: Config> = StorageMap<_, Twox64Concat, OrganizationIdOf<T>, BoundedApplicantsPerOrg<T>, ValueQuery>;
 
 	#[pallet::event]
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	pub enum Event<T: Config> {
-		/// Vision signed [AccountID, DaoIdOf]
+		/// Vision signed [AccountID, OrganizationIdOf]
```
Comment (stojanov-igor) [inline pallets/dao/src/lib.rs]: Why we need the Of? Why not `OrganizationId`

**Review by stojanov-igor — APPROVED** [review]

File: pallets/dao/src/lib.rs (hunk 3 above)
Comment (f-gate) [reply pallets/dao/src/lib.rs]: https://github.com/UniversalDot/universal-dot-node/pull/96/commits/f81cfcd7f65c9ad8c93a8876011ca6fd2c0252b7

File: pallets/dao/src/lib.rs (hunk 2 above)
Comment (f-gate) [reply pallets/dao/src/lib.rs]: https://github.com/UniversalDot/universal-dot-node/pull/96/commits/f81cfcd7f65c9ad8c93a8876011ca6fd2c0252b7

**Review by stojanov-igor — APPROVED** [review]

**Pull request merged**